### DISCUSSION TITLE
请问如何从 custom U-Boot layout 固件切换为 OpenWrt U-Boot layout 固件

### DISCUSSION THREAD

**Discussion by TomCzHen** [immortalwrt/immortalwrt#1483] [Q&A] [answered]
目前的设备是红米 AX6000,使用的是  custom U-Boot layout ，看到后续不会有这个layout的固件提供了，请问如何切换到 OpenWrt U-Boot layout。

```
root@AX6000:~# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00100000 00020000 "BL2"
mtd1: 00040000 00020000 "Nvram"
mtd2: 00040000 00020000 "Bdata"
mtd3: 00200000 00020000 "Factory"
mtd4: 00200000 00020000 "FIP"
mtd5: 00040000 00020000 "crash"
mtd6: 00040000 00020000 "crash_log"
mtd7: 06e00000 00020000 "ubi"
```

```
root@AX6000:~# cat /proc/partitions 
major minor  #blocks  name

  31        0       1024 mtdblock0
  31        1        256 mtdblock1
  31        2        256 mtdblock2
  31        3       2048 mtdblock3
  31        4       2048 mtdblock4
  31        5        256 mtdblock5
  31        6        256 mtdblock6
  31        7     112640 mtdblock7
 254        0       8060 ubiblock0_1
```

```
root@AX6000:~# ubinfo -a
UBI version:                    1
Count of UBI devices:           1
UBI control device major/minor: 10:127
Present UBI devices:            ubi0

ubi0
Volumes count:                           3
Logical eraseblock size:                 126976 bytes, 124.0 KiB
Total amount of logical eraseblocks:     880 (111738880 bytes, 106.5 MiB)
Amount of available logical eraseblocks: 0 (0 bytes)
Maximum count of volumes                 128
Count of bad physical eraseblocks:       0
Count of reserved physical eraseblocks:  19
Current maximum erase counter value:     12
Minimum input/output unit size:          2048 bytes
Character device major/minor:            249:0
Present volumes:                         0, 1, 2

Volume ID:   0 (on ubi0)
Type:        dynamic
Alignment:   1
Size:        42 LEBs (5332992 bytes, 5.0 MiB)
State:       OK
Name:        kernel
Character device major/minor: 249:1
-----------------------------------
Volume ID:   1 (on ubi0)
Type:        dynamic
Alignment:   1
Size:        65 LEBs (8253440 bytes, 7.8 MiB)
State:       OK
Name:        rootfs
Character device major/minor: 249:2
-----------------------------------
Volume ID:   2 (on ubi0)
Type:        dynamic
Alignment:   1
Size:        748 LEBs (94978048 bytes, 90.5 MiB)
State:       OK
Name:        rootfs_data
Character device major/minor: 249:3
```

**Comment by Fvpder** [comment]
ssh上去刷入openwrt官方版uboot，`mtd write /path/to/uboot/uboot.fip FIP`，然后进入uboot使用tftpd刷写基础内核就能正常进系统，再刷入sysupgrade固件就好了

**Comment by libiunc** [comment]
sir this way

https://blog.imouto.in/#/posts/10

https://drive.wrt.moe/uboot/mediatek

备份分区后直接刷*fip，炸了能用uartboot救砖。~~欢迎反馈错误~~

**Comment by TomCzHen** [comment, marked as answer]
参考了大佬们提供的资料，自己组合搞定了。

当前 uboot 是 [hanwckf 大佬的 mt798x-uboot](https://cmi.hanwckf.top/p/mt798x-uboot-usage/#%E7%BA%A2%E7%B1%B3ax6000%E5%A4%9A%E7%A7%8D%E5%88%86%E5%8C%BA%E5%B8%83%E5%B1%80%E5%9B%BA%E4%BB%B6%E5%85%BC%E5%AE%B9%E6%80%A7)，这个 uboot 不支持 ubootmod layout 的固件。参考了 [openwrt 的红米 ax6000 刷机教程](https://openwrt.org/toh/xiaomi/redmi_ax6000?s[]=xiaomi&s[]=redmi&s[]=router&s[]=ax6000&s[]=openwrt&s[]=boot&s[]=layout#installing_openwrt_ubootmod_layout)， 发现openwrt 教程刷 ubootmod layout 固件是先刷 stock layout 再刷的 ubootmod layout ，但是并没有 stock layout uboot 存在。

所以自己推测 mt798x-uboot stock 布局下 ubootmod factory 镜像是可以启动的，所以先切换 mt798x-uboot 到 stock 布局，刷入 ubootmod factory,  再根据官方刷机教程操作完成了刷机。

**Reply by lalasou** [reply]
你好 我现在也是hanwckf的uboot 如何 和他的源码编译的固件  如何切换到官方uboot？
直接在h大uboot下选择uboot (default)  然后刷入immortalwrt-23.05.4-mediatek-filogic-xiaomi_redmi-router-ax6000-ubootmod-initramfs-factory然后在刷如 immortalwrt-23.05.4-mediatek-filogic-xiaomi_redmi-router-ax6000-ubootmod-squashfs-sysupgrade对吗？


请问 https://firmware-selector.immortalwrt.org/?version=23.05.4&target=mediatek%2Ffilogic&id=xiaomi_redmi-router-ax6000
这里的 
Xiaomi Redmi Router AX6000 (custom U-Boot layout)  又是什么版本分区？？？

**Reply by BabXhwZUNBeTV** [reply]
以下为个人操作经验。先在 custom U-Boot layout 固件安装 kmod  mtd write, 配合 ssh 命令，通过 scp 上传文件。刷 FIP 和 BL2 ，然后 TFTP 启动 initramfs 固件，再 luci 刷 24 10 的固件。默认 TFTP 服务器地址 192.168.1.254 。

**Reply by lalasou** [reply]
> 以下为个人操作经验。先在 custom U-Boot layout 固件安装 kmod  mtd write, 配合 ssh 命令，通过 scp 上传文件。刷 FIP 和 BL2 ，然后 TFTP 启动 initramfs 固件，再 luci 刷 24 10 的固件。默认 TFTP 服务器地址 192.168.1.254 。


偶买噶  好复杂 
算了 就用闭源固件吧

**Comment by WFJ1987** [comment]
我的CT3003也是 不知道怎么切换，现在刷不了新版本

**Comment by SolidZORO** [comment]
刚从 23 刷到 24。随便说一下。

我 23 也是 h 大大的 custom U-Boot layout，长按路由器 reset 插入电源，进入 webUI。上传 24 的 uboot，可以刷 ubootmod-initramfs-factory.ubi 重启之后是救援模式，但是这里不能刷 ubootmod-squashfs-sysupgrade.itb 刷了就起不来。

后来我尝试刷 24 的 stock，也就是 immortalwrt-24.10.2-mediatek-filogic-xiaomi_redmi-router-ax6000-stock-initramfs-factory.ubi，在 webUI 选 default（不要选 100m），刷进去之后进入救援模式，然后在救援模式继续上传刷 immortalwrt-24.10.2-mediatek-filogic-xiaomi_redmi-router-ax6000-stock-squashfs-sysupgrade.bin 就可以了。

不过我用 iPerf3 测试了一下，在 23 我能跑到 1600M，但是 24 只能 900M。感觉亏大了～不管了，先这样，等 25 看看能不能优化好。

**Comment by SolidZORO** [comment]
h大大23：**1666MB**
<img width="756" height="744" alt="c1c7a2e95e8d08b95197ba8bc2a5991a" src="https://github.com/user-attachments/assets/539a69a2-52ec-4fa0-90bf-1d0a6ebd4b73" />



immortalwrt  官方stock 24： **920MB**
<img width="996" height="864" alt="image" src="https://github.com/user-attachments/assets/feacae3e-f1c6-4e09-b2d8-6efab4713a01" />


这个差距真的有点大。

**Reply by Fvpder** [reply]
H大用的闭源驱动，有mtk的硬件加速，immortalwrt是跟着官方主线走的，速度慢一点也正常

**Reply by SolidZORO** [reply]
> H大用的闭源驱动，有mtk的硬件加速，immortalwrt是跟着官方主线走的，速度慢一点也正常

是啊，我也坦然了。毕竟没有太多机会用到 1000M。反倒是闭源驱动在我 iPhone12 上最高只能 480M，但是在 v24 上能跑到 860M，吓我一跳。

**Reply by Fvpder** [reply]
闭源驱动有好多版本的固件用iPhone只能跑到1t1r，我最近换了237大佬的6.6内核固件，wifi6的iphone能正常协商到2t2r，但是wifi5的iphone还是只能1t1r跑433

**Reply by SolidZORO** [reply]
> 闭源驱动有好多版本的固件用iPhone只能跑到1t1r，我最近换了237大佬的6.6内核固件，wifi6的iphone能正常协商到2t2r，但是wifi5的iphone还是只能1t1r跑433

我现在 24 wifi 非常不稳定，考虑换回闭源固件了。

**Comment by SolidZORO** [comment]
更新一下，wifi 全面不行，我又从 24 降回到 23 了。

我电视机角度比较刁钻，之前200m，现在才 20m…… 太拉垮了。

相对于版本，我觉得稳定性最重要。不推荐追新了。
